api/src/main/java/org/openmrs/module/kenyaemr/reporting/builder/cacx/CACXRegisterReportBuilder.java

```diff
@@ -0,0 +1,149 @@
+        dsd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        dsd.addParameter(new Parameter("endDate", "End Date", Date.class));
+
+        String paramMapping = "startDate=${startDate},endDate=${endDate}";
+
+        DataConverter nameFormatter = new ObjectFormatter("{familyName}, {givenName} {middleName}");
+        DataDefinition nameDef = new ConvertedPersonDataDefinition("name", new PreferredNameDataDefinition(), nameFormatter);
+        PatientIdentifierType upn = MetadataUtils.existing(PatientIdentifierType.class, HivMetadata._PatientIdentifierType.UNIQUE_PATIENT_NUMBER);
+        DataConverter identifierFormatter = new ObjectFormatter("{identifier}");
+        DataDefinition identifierDef = new ConvertedPatientDataDefinition("identifier", new PatientIdentifierDataDefinition(upn.getName(), upn), identifierFormatter);
+
+        PersonAttributeType phoneNumber = MetadataUtils.existing(PersonAttributeType.class, CommonMetadata._PersonAttributeType.TELEPHONE_CONTACT);
+
+        dsd.addColumn("id", new PatientIdDataDefinition(), "");
+        dsd.addColumn("Visit Date", new EncounterDatetimeDataDefinition(),"", new DateConverter(ENC_DATE_FORMAT));
+        dsd.addColumn("Visit Type", new CACXVisitTypeDataDefinition(), null);
+        dsd.addColumn("Name", nameDef, "");
+        dsd.addColumn("ID Number", new CACXNationalIdDataDefinition(), "");
+        dsd.addColumn("Sex", new GenderDataDefinition(), "");
+        dsd.addColumn("Phone Number", new PersonAttributeDataDefinition(phoneNumber), "");
+        dsd.addColumn("Age in years", new AgeDataDefinition(), "");
+
+        CACXHivStatusDataDefinition cacxHivStatusDataDefinition = new CACXHivStatusDataDefinition();
+        cacxHivStatusDataDefinition.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        cacxHivStatusDataDefinition.addParameter(new Parameter("endDate", "End Date", Date.class));
+        dsd.addColumn("HIV Status", cacxHivStatusDataDefinition, paramMapping, null);
+
+        CACXPopulationTypeDataDefinition cacxPopulationTypeDataDefinition = new CACXPopulationTypeDataDefinition();
+        cacxPopulationTypeDataDefinition.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        cacxPopulationTypeDataDefinition.addParameter(new Parameter("endDate", "End Date", Date.class));
+        dsd.addColumn("Population Type", cacxPopulationTypeDataDefinition, paramMapping, null);
+
+        dsd.addColumn("County of Residence", new CACXCountyDataDefinition(), null);
+
+        CACXScreeningResultsDataDefinition cacxScreeningResultsDataDefinition = new CACXScreeningResultsDataDefinition();
+        cacxScreeningResultsDataDefinition.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        cacxScreeningResultsDataDefinition.addParameter(new Parameter("endDate", "End Date", Date.class));
+        dsd.addColumn("Screening Modalities", cacxScreeningResultsDataDefinition, paramMapping, null);
```

Comment:
Please confirm if this shouldn't pick recordings on the encounter date, and should be blank if nothing was recorded. If that's the case then we don't need the start and end date params. You simply join to the cohort using encounter_id